src/main/java/org/asciidoc/intellij/lexer/asciidoc.flex

```diff
@@ -400,19 +400,21 @@ LINKEND = "]"
   [^]                  { return AsciiDocTokenTypes.REF; }
 }
 
-<LINKFILE> {
-  "#"                  { yybegin(LINKANCHOR); return AsciiDocTokenTypes.LINKANCHOR; }
+<LINKFILE, LINKANCHOR> {
   {LINKTEXT_START}     { yybegin(LINKTEXT); return AsciiDocTokenTypes.LINKTEXT_START; }
+}
+
+<LINKFILE> {
+  "#"                  { yybegin(LINKANCHOR); return AsciiDocTokenTypes.SEPARATOR; }
```

Comment:
Instest of having "#" as part of the LINKTEXT_START token, I keep the "#" separate, as the "#" is not part of the ID